store/boltdbStore/business.go

```diff
@@ -18,52 +18,217 @@
+func (bs *businessStore) UpdateBusiness(b *model.Business) error {
+	if b.ID == "" || b.Name == "" || b.Owner == "" {
+		log.Errorf("[Store][database] update business missing some params: %+v", b)
+		return errors.New("Update Business missing some params")
+	}
+
+	dbOp := bs.handler
+
+	b.ModifyTime = time.Now()
+
+	lock := bs.lock
+	lock.Lock()
+	defer lock.Unlock()
+
+	if err := dbOp.SaveValue(DataTypeBusiness, b.ID, b); err != nil {
+		log.Errorf("[Store][database] add business err : %s", err.Error())
+		return store.Error(err)
+	}
+
 	return nil
 }
 
-// 更新业务集token
-func (b *businessStore) UpdateBusinessToken(bid string, token string) error {
-	//TODO
-	return nil
+// UpdateBusinessToken 更新业务集token
+func (bs *businessStore) UpdateBusinessToken(bid string, token string) error {
+	if bid == "" || token == "" {
+		log.Errorf("[Store][business] update business token missing some params")
+		return errors.New("Update Business Token missing some params")
+	}
+
+	// first: get business by bid
+	b, err := bs.GetBusinessByID(bid)
+	if err != nil {
+		return store.Error(err)
+	}
+
+	b.Token = token
+	return bs.UpdateBusiness(b)
 }
 
-// 查询owner下业务集
-func (b *businessStore) ListBusiness(owner string) ([]*model.Business, error) {
-	//TODO
-	return nil, nil
+// ListBusiness 查询owner下业务集
+func (bs *businessStore) ListBusiness(owner string) ([]*model.Business, error) {
+	if owner == "" {
+		log.Errorf("[Store][business] list business missing owner")
+		return nil, errors.New("List Business Mising param owner")
+	}
+
+	result, err := bs.listAllBusiness()
+	if err != nil {
+		return nil, err
+	}
+
+	ans := make([]*model.Business, 0)
+
+	for i := range result {
+		record := result[i]
+		if strings.Contains(owner, record.Owner) {
+			ans = append(ans, record)
+		}
+	}
+
+	return ans, nil
 }
 
-// 根据业务集ID获取业务集详情
-func (b *businessStore) GetBusinessByID(id string) (*model.Business, error) {
-	//TODO
-	return nil, nil
+// GetBusinessByID 根据业务集ID获取业务集详情
+func (bs *businessStore) GetBusinessByID(id string) (*model.Business, error) {
+
+	if id == "" {
+		log.Errorf("[Store][database] get business missing id")
+		return nil, errors.New("Get Business missing some params")
+	}
+
+	lock := bs.lock
+	lock.RLock()
+	defer lock.RUnlock()
+
+	return bs.lockfreeGetBusinessByID(id)
 }
 
-// 根据mtime获取增量数据
-func (b *businessStore) GetMoreBusiness(mtime time.Time) ([]*model.Business, error) {
-	//TODO
+// lockfreeGetBusinessByID 根据业务集ID获取业务集详情
+func (bs *businessStore) lockfreeGetBusinessByID(id string) (*model.Business, error) {
+	dbOp := bs.handler
+
+	result, err := dbOp.LoadValues(DataTypeBusiness, []string{id}, &model.Business{})
+	if err != nil {
+		return nil, store.Error(err)
+	}
+
+	val := result[id]
+	if val != nil {
+		return val.(*model.Business), nil
+	}
+
 	return nil, nil
-}
\ No newline at end of file
+}
+
+// GetMoreBusiness 根据mtime获取增量数据
+func (bs *businessStore) GetMoreBusiness(mtime time.Time) ([]*model.Business, error) {
+	result, err := bs.listAllBusiness()
```

Comment:
you shouldn't load all values here and filter it, you can use the LoadValuesByFilter to do the filtering

Reply:
already fix